### DISCUSSION TITLE
How to generate the index.json file for search?

### DISCUSSION THREAD

**Discussion by clothstir** [Lednerb/bilberry-hugo-theme#404] [Q&A] [answered]
Greetings! I have the search set to true in my config and have entered the API key, etc. My site generates properly with no errors. However, I don't understand how to generate the index.json file...nothing seems to be happening. Except for things like the title, etc., the only thing I've changed in the config is to comment out the entire [languages] block (I don't understand what "delete the [Languages] blog" means and I am only using English). What am I missing?

**Comment by igor-baiborodine** [comment]
Here it is: 

![image](https://user-images.githubusercontent.com/1027701/173809041-a012ec3c-b491-4d1c-9379-a688c98b04a5.png)

![image](https://user-images.githubusercontent.com/1027701/173809728-46371f63-0d11-4323-b689-5d484d225c6d.png)

**Reply by clothstir** [reply]
Hi. I'm aware of the instructions. As I said, my site generates properly with no errors, which means I have run the Hugo command. Many times. My question is why the index.json file is not being generated when I do.

**Reply by igor-baiborodine** [reply]
Could you please provide me with the link to the GitHub repo for your website? Thus, I can try testing it in my local dev.

**Comment by Lednerb** [comment]
@clothstir 
>  the only thing I've changed in the config is to comment out the entire [languages] block

I think the problem occurs because of the inactive translations block.

Actually it's needed when the `index.json` file is generated:
https://github.com/Lednerb/bilberry-hugo-theme/blob/fe4bc1b51a5d3a2b9727d3e6d26bea7010b756fa/layouts/index.json#L45-L47


As a workaround please let the language block active in your `config.toml` file.
Is there a specific reason why you've disabled that?

If so: Enable the langauge block for generating the `index.json` and disable it again after the command has generated the needed data.

**Comment by igor-baiborodine** [comment, marked as answer]
Comments in the `config.yml` file were updated:
![image](https://user-images.githubusercontent.com/1027701/173923571-a0b4519b-cfe9-429f-a3d2-ec4f79649822.png)

**Comment by clothstir** [comment]
That appears to have been the problem! Thank you. I see you changed the language in that section to help slow people like me :)
